enterprise/templates/enterprise/grant_data_sharing_permissions.html

```diff
@@ -31,7 +31,11 @@
                         </h2>
                     </div>
                     <span class="text">
-                        ${_("In order to link your account, we need consent to share your course data with {provider}.").format(provider=sso_provider)}
+                        % if not course_id:
+                            ${_("In order to link your account, we need consent to share your course data with {provider}.").format(provider=ec_name)}
+                        % else:
+                            ${_("In order to access {course_name}, we need consent to share your course data with {provider}.").format(provider=ec_name, course_name=course_name)}
```

Comment:
It depends on the configuration for a given Enterprise / Identity Provider -- it's possible to force people to provide consent when attempting to access the system through the SSO door.